aten/src/ATen/ParallelMobileTPool.h

```diff
@@ -0,0 +1,19 @@
+#ifdef C10_MOBILE
+#pragma once
+#include "caffe2/utils/threadpool/ThreadPool.h"
+#include "caffe2/utils/threadpool/pthreadpool.h"
+
+namespace at {
+namespace {
+static std::unique_ptr<caffe2::ThreadPool> thread_pool_ = nullptr;
```

Comment:
I think you need move these to cpp files otherwise each compile unit will have their own instances...